src/thunderbots/software/ai/hl/stp/action/movespin_action.cpp

```diff
@@ -8,12 +8,14 @@ MoveSpinAction::MoveSpinAction(double close_to_dest_threshold)
 }
 
 std::unique_ptr<Intent> MoveSpinAction::updateStateAndGetNextIntent(
-    const Robot& robot, Point destination, AngularVelocity angular_velocity)
+    const Robot& robot, Point destination, AngularVelocity angular_velocity,
+    double final_speed)
```

Comment:
Could we make `final_speed` here an `AngularVelocity` to be consistent? Same goes for the other cases in this PR where we're dealing with angular values.

Reply:
This is the final speed for the movement of the robot, not the rotation. It's the same as the final speed for the MoveAction

Reply:
I interpreted `final_speed` as linear speed rather than rotation speed. Would changing the name of `final_speed` to something like `final_linear_speed` help the understanding?

Reply:
Ah, I see, yes that naming change would def. help.